src/main/java/nextstep/courses/domain/session/Session.java

```diff
@@ -0,0 +1,86 @@
+package nextstep.courses.domain.session;
+
+import nextstep.courses.dto.SessionPaymentInfo;
+import nextstep.courses.type.SessionState;
+import nextstep.payments.domain.Payment;
+
+import java.time.LocalDateTime;
+import java.util.Objects;
+
+public abstract class Session {
+
+    static final int NOT_ASSIGNED = -1;
+
+    private Long id;
+    private final String name;
+    private final CoverImage coverImage;
+
+    protected int enrollment;
+    protected final int maxEnrollment;
+    protected final long sessionFee;
+
+    protected SessionState sessionState;
+
+    protected LocalDateTime startDate;
+    protected LocalDateTime endDate;
+
+    protected Session(String name, CoverImage coverImage, int maxEnrollment, SessionState sessionState,
+                      long sessionFee, LocalDateTime startDate, LocalDateTime endDate) {
+        this((long) NOT_ASSIGNED, name, coverImage, maxEnrollment, sessionState, sessionFee, startDate, endDate);
+    }
+
+    protected Session(Long id, String name, CoverImage coverImage, int maxEnrollment,
+                      SessionState sessionState, long sessionFee, LocalDateTime startDate, LocalDateTime endDate) {
+        this.id = id;
+        this.name = name;
+        this.coverImage = coverImage;
+        this.maxEnrollment = maxEnrollment;
+        this.sessionFee = sessionFee;
+        this.sessionState = sessionState;
+        this.startDate = startDate;
+        this.endDate = endDate;
+    }
+
+    public final SessionPaymentInfo tryRegisterForSession() {
+        if (sessionState != SessionState.OPEN) {
```

Comment:
```suggestion
        if (!SessionState.canRegister()) {
```

과 같이 값을 꺼내지 않고 메세지를 보내는건 어떨까요?

Reply:
객체에 메세지 보내는 방식으로 변경했습니다!